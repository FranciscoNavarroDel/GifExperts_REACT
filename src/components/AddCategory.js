import React from "react";
import { useState } from "react";
import PropTypes from "prop-types";


const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState(
    "Escriba una categorÃ­a a agregar"
  );

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim()) {
      setCategories((categories) => [ inputValue, ...categories]);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      ></input>
    </form>
  );
};


AddCategory.propTypes={
    setCategories:PropTypes.func.isRequired
}

export { AddCategory };
