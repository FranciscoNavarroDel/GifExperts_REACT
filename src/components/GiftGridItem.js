import React from 'react'
import PropTypes from 'prop-types'

export const GiftGridItem = ({id,title,gif}) => {

    return (
        <div key={id} className="card animate__animated animate__rotateIn ">
            <img src={gif} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
}
