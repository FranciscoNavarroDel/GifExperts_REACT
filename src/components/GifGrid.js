import React from "react";
import { useState, useEffect } from "react";
import { GiftGridItem } from "./GiftGridItem";
import { getGifs } from "../helpers/getGifs";



export const GifGrid = ({ category }) => {
  // Hace que el codigo sea renderizado solo la primera vez
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then( setImages)
  }, [category]);



  return (
    <>
      <h3>{category}</h3>
      <div className="card_grid">
        {images.map((img) => (
          <GiftGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
