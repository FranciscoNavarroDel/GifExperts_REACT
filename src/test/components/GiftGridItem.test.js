import { shallow } from "enzyme"
import React from "react"
import { GiftGridItem } from "../../components/GiftGridItem"

  

describe("Prueba al item GiftGridItem",()=>{
   const id=1
   const title="probandoItem"
   const gif="http//probando.cl"
   const wrapper= shallow(<GiftGridItem id={id} title={title} gif={gif} />)

 test('debe mostrar el componente correctamente ', () => {

    expect(wrapper).toMatchSnapshot()
 })
 
 test("debe tener un parrafo con el title",()=>{
   
   const parrafo = wrapper.find("p")
   expect(parrafo.text().trim()).toBe(title)

 })

 test("debe tener la imagen al url y alt de los props",()=>{
   
   const img = wrapper.find("img")
   const testProps= {
      src:gif,
      alt:title,
   }

   expect(img.props()).toEqual(testProps)

 })

 test('debe tener animate__rotateIn', () => {
   const div = wrapper.find("div") 
   const className=div.prop("className")

   expect(className.includes("animate__rotateIn")).toBe(true)

 })
 


})