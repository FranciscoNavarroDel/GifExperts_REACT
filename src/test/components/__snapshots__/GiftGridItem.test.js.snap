// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba al item GiftGridItem debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__rotateIn "
  key="1"
>
  <img
    alt="probandoItem"
    src="http//probando.cl"
  />
  <p>
    probandoItem
  </p>
</div>
`;
